### PR TITLE
GEODE-4339: cpp ssl example

### PR DESCRIPTION
**Pull request by mmartell**

### CODE REVIEW

File: cppcache/src/TcrConnection.cpp [hunk 1]
```diff
@@ -568,9 +568,9 @@ Connector* TcrConnection::createConnection(
                                .getSystemProperties();
   if (systemProperties.sslEnabled()) {
     socket = new TcpSslConn(endpoint, connectTimeout, maxBuffSizePool,
-                            systemProperties.sslKeystorePassword().c_str(),
                             systemProperties.sslTrustStore().c_str(),
-                            systemProperties.sslKeyStore().c_str());
+                            systemProperties.sslKeyStore().c_str(),
+                            systemProperties.sslKeystorePassword().c_str());
```
Comment (pdxcodemonkey) [inline cppcache/src/TcrConnection.cpp]: Whitespace issue here?  This diff looks weird...

File: examples/cpp/sslputget/main.cpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+using apache::geode::client::CacheableString;
+using apache::geode::client::CacheFactory;
+using apache::geode::client::RegionShortcut;
+
+std::string myGetcwd() {
+  char buf[PATH_MAX];
+  std::string cwd;
+
+#ifdef _MSC_VER
```
Comment (pdxcodemonkey) [inline examples/cpp/sslputget/main.cpp]: If we changed this to something like:
```
#ifdef _MSC_VER
  char buf[MAX_PATH];
  if (_getcwd(buf, MAX_PATH)) {
    cwd = buf;
  }
#else
  char buf[PATH_MAX];
  if (getcwd(buf, PATH_MAX)) {
    cwd = buf;
  }
#endif
```
could we get rid of all the awful _MSC_VER stuff above?

File: cppcache/src/TcrConnection.cpp (hunk 1 above)
Comment (mmartell) [reply cppcache/src/TcrConnection.cpp]: Hmm. Not seeing the issue here.

File: examples/cpp/sslputget/main.cpp (hunk 2 above)
Comment (mmartell) [reply examples/cpp/sslputget/main.cpp]: Agreed we can make this a little cleaner. (Ultimately we'll go to std::filesystem)

**Pull request merged**